src/model/formula/Formula.Disjunction.js

```diff
@@ -58,6 +58,16 @@ class Disjunction extends Formula {
     this.subLeft.setVariable(from, to);
     this.subRight.setVariable(from, to);
   }
```

Comment:
`setVariable` is a confusing name for a very standard operation – substitution.

So call this `substitute` and allow any term to be substituted for the variable, not just another variable.

It’s also bad practice to modify the original formula by substitution. Make a new copy on the fly, e.g., in this case:
```javascript
substitute: (var, term) =>
  new Disjunction(
    this.subLeft.substitute(var, term),
    this.subRight.substitute(var, term)
  );
```
If all `substitute` methods create new copies, then this is safe and will create a completely new formula.

Make sure that also constants and quantified formulas that quantify `var` return a new copy of themselves in `substitute`.

Reply:
I don't understand what you are trying to say here: "So call this substitute and allow any term to be substituted for the variable, not just another variable.". Are you implying that Constants should be substituted as well?